docs/DeveloperGuide.md

```diff
@@ -2,32 +2,143 @@
+
+
+### [Proposed]  Check borrowing status
+Description: Check status of borrowed book  
+Format: status -title TITLE
+Example:
+- status -title C++Primer  
+The program will output the details of the relevant book being borrowed and also show the due date of the loan. The program will handle cases where there is no such book in the borrow history.
+
+
+### [Proposed]  Return book
+Description: Return a book to the library  
+Format: return -title TITLE
+Example:
+- return -title C++Primer
+For successful returns of books, the program will output a string showing that the action is successful and mark the book as available for borrowing by other users in the system. The program will handle cases of incorrect title input and unsuccessful returns.
+
+
+### [Proposed] Add book
+![](images/Inventory%20class%20diagram.png)
+Description: Add new books into the system  
+Format: librarian -title TITLE -topic TOPIC -author AUTHOR -isbn ISBN -action add
+Example:
+- librarian -title C++Primer -topic Programming -author James -isbn 12345 -action add  
+This feature is only applicable for admin. For successful addition of new book, the program will output a message to inform the librarian. The new book will also be added into the inventory of the system so that users can now borrow the book from the library. Error inputs and missing inputs will be handled accordingly.
+ 
+
+### [Proposed] Delete book
+Description: Remove books from the system  
+Format: librarian -title TITLE -topic TOPIC -author AUTHOR -isbn ISBN -action delete
+Example:
+- librarian -title C++Primer -topic Programming -author James -isbn 12345 -action delete  
+This feature is only applicable for admin. For successful deletion of book, the program will output a message to inform the librarian. The book will also be deleted from the inventory of the system so that it will not be able to be borrowed by users anymore. Error inputs and missing inputs will be handled accordingly.
+
+
+### [Proposed] List book
+Description: List all the books in the library  
+Format: list
+Example:
+- list  
+The program will list out all the books in the library inventory. If the inventory is empty, the program will output a message indicating that there is currently no book in the library system.
+
+
+### [Proposed] View history
+Description: Check the borrow history of the books (admin and user versions). The admin version will allow admin to access the whole borrow history, while users can only view their own borrow history.  
+Format: history
+Example:
+- history  
+The program will output the borrow history of books according to the accessibility as mentioned in the description above. The output includes the book title and other book details, the status of the book (either it is available for borrowing or it has been borrowed at the moment), and also details of the loan, which include the date and time of borrowing and returning.
```

Comment:
It will be helpful to include which classes are involved in this feature. Using a sequence diagram is probably a good way to do that.